packaging/common/substitute.pl

```diff
@@ -34,16 +31,16 @@ sub read_file {
 	# $1 is the file name and must exist.
 	my $contents;
 	my $file = "$_[0]";
-	my $f_handle;
 	# Autogen has been run, the file will be there.
-	if (-e $file) {
+	if (-s $file) {
+            local $/;
```

Comment:
overriding with a "undefined" $/ is how you get "the whole file is one line" to work